cmd/cli/kubectl-kyverno/commands/test/output.go

```diff
@@ -191,8 +191,13 @@ func printTestResult(
 				for _, rule := range lookupRuleResponses(test, response.PolicyResponse.Rules...) {
 					// perform test checks
 					ok, message, reason := checkResult(test, fs, resoucePath, response, rule)
-					// if checks failed but we were expecting a fail it's considered a success
-					success := ok || (!ok && test.Result == policyreportv1alpha2.StatusFail)
+					var success bool
+					if reason == "Want fail, got pass" {
```

Comment:
This is not the right way to solve this, as matching the reason string is a bad approach, I think the flaw is in how we calculate `ok` and `reason`, and the right solution would be to fix that.